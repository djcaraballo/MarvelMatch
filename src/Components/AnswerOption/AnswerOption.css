@import url('https://fonts.googleapis.com/css?family=Fjalla+One|Orbitron');

li {
  list-style-type: none;
  color: #fff;
  font-family: 'Orbitron', sans-serif;
  font-size: 1.5em;
  line-height: 1.5em;
}

.radio-button {
  margin: 10px;
  font-size: 2.5em;
  vertical-align: text-bottom;
}

.radio-button:hover {
  cursor: pointer;
}