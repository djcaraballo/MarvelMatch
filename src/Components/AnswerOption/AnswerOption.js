import React from 'react'
import PropTypes from 'prop-types'

const AnswerOption = (props) => {
  const { answerType, answerContent, answer, onAnswerSelected } = props
  return (
    <li>
      <input
        type='radio'
        className='radio-button'
        name='radio-group'
        checked={answerType === answer}
        id={answerType}
        value={answerType}
        disabled={answer}
        onChange={onAnswerSelected}
      />
      <label className='radio-label' htmlFor={answerType}>
        {answerContent}
      </label>
    </li>
  )
}

AnswerOption.propTypes = {
  answerType: PropTypes.string,
  answerContent: PropTypes.string,
  answer: PropTypes.string,
  onAnswerSelected: PropTypes.func,
}

export default AnswerOption

