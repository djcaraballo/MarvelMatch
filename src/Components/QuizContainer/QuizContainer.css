@import url('https://fonts.googleapis.com/css?family=Fjalla+One|Orbitron');

.answer-options {
  margin: 2em auto 10em;
  padding: 0;
}

