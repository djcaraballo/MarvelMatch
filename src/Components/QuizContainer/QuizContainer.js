import React from 'react'
import PropTypes from 'prop-types'
import Question from '../Questions/Questions'
import QuestionCount from '../QuestionCount/QuestionCount'
import AnswerOption from '../AnswerOption/AnswerOption'

const QuizContainer = (props) => {
  const {
    answer,
    answerOptions,
    counter,
    question,
    questionId,
    questionTotal,
    onAnswerSelected,
  } = props
  
  const renderAnswerOptions = (key) => {
    return (
      <AnswerOption
          key={key.content}
          answerContent={key.content}
          answerType={key.type}
          answer={answer}
          questionId={questionId}
          onAnswerSelected={onAnswerSelected}
      />
    )
  }

  return (
    <div className='quiz-container'>
      <QuestionCount
        counter={questionId}
        total={questionTotal}
      />
      <Question text={question} />
      <ul className='answer-options'>
        {answerOptions.map(renderAnswerOptions)}
      </ul>
    </div>
  )
}

QuizContainer.propTypes = {
  answer: PropTypes.string,
  answerOptions: PropTypes.array,
  counter: PropTypes.number,
  question: PropTypes.string,
  questionId: PropTypes.number,
  questionTotal: PropTypes.number,
  onAnswerSelected: PropTypes.func
}

export default QuizContainer


