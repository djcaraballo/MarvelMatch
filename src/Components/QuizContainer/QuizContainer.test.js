import React from 'react'
import QuizContainer from './QuizContainer'
import AnswerOption from '../AnswerOption/AnswerOption'
import { shallow } from 'enzyme'

describe('QuizContainer Component', () => {
  let wrapper
  let mockAnswer = ''
  let mockAnswerOptions = []
  let mockCounter = 0
  let mockQuestion = ''
  let mockQuestionId = 0
  let mockQuestionTotal = 0
  let mockOnAnswerSelected = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <QuizContainer 
        answer={mockAnswer}
        answerOptions={mockAnswerOptions}
        counter={mockCounter}
        question={mockQuestion}
        questionId={mockQuestionId}
        questionTotal={mockQuestionTotal}
        onAnswerSelected={mockOnAnswerSelected}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('renderAnswerOptions function', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallow(
        <QuizContainer 
          answer={mockAnswer}
          answerOptions={mockAnswerOptions}
          counter={mockCounter}
          question={mockQuestion}
          questionId={mockQuestionId}
          questionTotal={mockQuestionTotal}
          onAnswerSelected={mockOnAnswerSelected}
        />
      )
    })

    xit('should return an AnswerOption component instance', () => {
      const mockKey = {type: '', content: ''}
      const expected = <AnswerOption
                          key={mockKey.content}
                          answerContent={mockKey.content}
                          answerType={mockKey.type}
                          answer={mockAnswer}
                          questionId={mockQuestionId}
                          onAnswerSelected={mockOnAnswerSelected} 
                        />
      wrapper.instance().renderAnswerOptions(mockKey)
      expect(wrapper.instance().renderAnswerOptions(mockKey)).toEqual(expected)
    })
  })
})