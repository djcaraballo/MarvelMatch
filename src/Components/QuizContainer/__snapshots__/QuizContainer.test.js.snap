// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizContainer Component should match the snapshot 1`] = `
<div
  className="quiz-container"
>
  <QuestionCount
    counter={0}
    total={0}
  />
  <Question
    text=""
  />
  <ul
    className="answer-options"
  />
</div>
`;
